### DISCUSSION TITLE
/dev/serial/by-id/ disappear when container restart

### DISCUSSION THREAD

**Discussion by barrydouu** [zwave-js/zwave-js-ui#3682] [Q&A]
Hello

I run zwave js ui in docker container from a long time. From some time, can't remember when, I have trouble restarting container. The /dev/serial/by-id disappear. It happens specially when I update zwavejsui (docker compose pull, docker compose up -d). Zwavejsui can't restart. I need to reboot my computer, and the docker start well.

Looking from the debug log of the driver, it happens after a soft reset of the controller (gen 5+)

I found that a software reset on gen 5+ is like unplugged replug the usb device.
But, it seems that udev don't detect it, and don't map /dev/serial/by-id again

I don't have any idea what to check. I've seen that we can disable soft reset, but if prefere to find what is the main root.

Thank you for your ideas / your help

**Comment by robertsLando** [comment]
Follow instructions here: https://github.com/zwave-js/node-zwave-js/issues/6341

If nothing works just disable soft reset in settings :)

**Reply by barrydouu** [reply]
Hello

Thanks you for the reply. I found this thread after I post here.

I have followed it.
I already use /dev/serial/by-id on the main Host (i use docker, on a raspberry pi 4, whith bookworm).
I use firmware 1.2
No esxi
No passthru with docker too
I have not tried serial over TCP. I will do it. But as the /dev/serial/... disappear on the main host, I think I will have the same trouble with it.

So I will disable soft reset. 

One last question : is there any guide to debug what happens with USB port , and udev ? Just to search if no other solution exist, on the os configuration.

**Reply by robertsLando** [reply]
`dmesg` should output info about that

**Reply by barrydouu** [reply]
Here is interesting log
kernel: usb 3-1.3: USB disconnect, device number 8
kernel: usb 3-1.3: new full-speed USB device number 9 using dwc2
kernel: usb 3-1.3: device descriptor read/64, error -71
kernel: usb 3-1.3: device descriptor read/64, error -71
kernel: usb 3-1.3: new full-speed USB device number 10 using dwc2
kernel: usb 3-1.3: device descriptor read/64, error -71
kernel: usb 3-1.3: device descriptor read/64, error -71
kernel: usb 3-1-port3: attempt power cycle
kernel: usb 3-1.3: new full-speed USB device number 11 using dwc2
kernel: usb 3-1.3: device not accepting address 11, error -71
kernel: usb 3-1.3: new full-speed USB device number 12 using dwc2
kernel: usb 3-1.3: device not accepting address 12, error -71
kernel: usb 3-1-port3: unable to enumerate USB device

It happens during container restart. So after soft reset, the host cannot reconnect usb device.
I'll try another port when I can access to the computer (it's a remote installation)
